sw/host/opentitantool/src/command/version.rs

```diff
@@ -15,19 +15,29 @@ pub struct Version {}
 
 #[derive(Debug, serde::Serialize)]
 pub struct VersionResponse {
-    pub version: String,
-    pub revision: String,
-    pub clean: bool,
-    pub timestamp: i64,
+    pub version: Option<String>,
+    pub revision: Option<String>,
+    pub clean: Option<bool>,
+}
+
+fn parse_variable(content: &str) -> Option<String> {
```

Comment:
This feels like a hack. Could you make it so that if stamping is disabled, the stamp-env is not given at all? Then you can just use `std::option_env!(...)`.

Reply:
My understanding is that this is not possible with the way rules_rust handles `rustc_env_file`: stamping is handled by replacing `{BUILD_SCM_REVISION}` and others literally in the env file. If stamping is disabled, the replacement does not occur. But the content of the `stamp-env.txt` file does not depend on whether stamping is enabled or not. This, admitedly, make the way stamping is handled by rustc not so useful.
We could add custom feature to our rules_rust fork so that writing "{BUILD_SCM_REVISION?}" deletes the variables if not present. Maybe this could even be accepted upstream?

Alternatively, this could be achieved with some custom bazel rule to produce either an empty file or a version file, depending on whether stamping is enabled or not, and then give the target to `rustc_env_file`, but this could be more trouble than it is worth. Something like:
```python
version_file_or(
    name = "ott_version_file",
    default_file = "//path/to/empty/file",
)
genrule(
  name = "ott_env_file",
  src = [":ott_version_file"],
  outs = ["stamp-env.txt"],
  cmd = "<inset awk script>", # convert from "KEY VALUE" to "KEY=VALUE"
)
```